lib/inertia_rails/renderer.rb

```diff
@@ -30,21 +27,30 @@ def initialize(component, controller, request, response, render_method, props: n
     end
 
     def render
+      set_vary_header
+
+      validate_partial_reload_optimization if rendering_partial_component?
+
+      return render_inertia_response if @request.headers['X-Inertia']
+      return render_ssr if configuration.ssr_enabled rescue nil
+
+      render_full_page
```

Comment:
Moved these into methods purely for readability. IMO this method should be super readable since it spells out the core functionality